utils/datasets_util.py

```diff
@@ -0,0 +1,259 @@
+import numpy as np
+import os
+import os.path as op
```

Comment:
Minor thing, but `op` also sounds like a short from `operation`. Can be a bit confusing.